sdk/analytics/src/main/java/com/klaviyo/analytics/model/Event.kt

```diff
@@ -17,18 +17,24 @@ class Event(val type: EventType, properties: Map<EventKey, Serializable>?) :
 
     constructor(type: String) : this(type, null)
 
-    fun setValue(value: String) = apply { this.value = value }
-    var value: String
-        get() = (this[EventKey.VALUE] ?: "") as String
+    fun setValue(value: Double) = apply { this.value = value }
+    var value: Double?
+        get() = when (val value = this[EventKey.VALUE]) {
+            is Double -> value
+            else -> try {
```

Comment:
Alternative would be to have a case for each Int, Float, etc.. this just looks cleaner 🤷